activerecord/lib/active_record/persistence.rb

```diff
@@ -357,10 +357,11 @@ def increment(attribute, by = 1)
     # +update_counters+, see that for more.
     # Returns +self+.
     def increment!(attribute, by = 1, touch: nil)
-      increment(attribute, by)
-      change = public_send(attribute) - (attribute_in_database(attribute.to_s) || 0)
-      self.class.update_counters(id, attribute => change, touch: touch)
-      clear_attribute_change(attribute) # eww
+      attribute_name = attribute_aliases.fetch(attribute.to_s) { attribute }.to_s
```

Comment:
It would be good to move this logic to CounterCache#update_counters method so that it can be used by everyone else who calls `update_counters`